controllers/users/register_v2.js

```diff
@@ -28,47 +28,88 @@ const registerV2 = async (req, res) => {
-
-      return {
-        user,
-        anonymousUser,
-        topics: topicRegistered,
-        locations
-      };
-    });
-  } catch (e) {
-    console.log('error on sql transaction', e);
-    return res.status(500).json({
-      status: 'error on sql transaction',
-      code: 500,
-      message: e
-    });
+  if (isUnverifiedUser) {
+    let updatedUser;
+    try {
+      users.verified_status = 'VERIFIED';
+      await checkUser.update(users);
+      updatedUser = await checkUser.save();
+    } catch (e) {
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+
+    const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics);
+    const locations = await LocationFunction.getAllLocationByIds(Locations, local_community);
+
+    insertedObject = {
+      user: updatedUser,
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, updatedUser.user_id, {
+        raw: false
+      }),
+      topics: topicRegistered,
+      locations
+    };
+  } else {
+    /**
+     * Inserting data to Postgres DB
+     */
+    try {
+      insertedObject = await sequelize.transaction(async (t) => {
+        const user = await UsersFunction.register(User, users, t);
+        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
+
+        const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics, t, true);
+        const locations = await LocationFunction.getAllLocationByIds(Locations, local_community, t);
+
+        return {
+          user,
+          anonymousUser,
+          topics: topicRegistered,
+          locations
+        };
+      });
+    } catch (e) {
+      console.log('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+    /**
+     * Inserting data to Postgres DB (END)
+     */
   }
-  /**
-   * Inserting data to Postgres DB (END)
-   */
 
   /**
    * Creating User to Getstream
    */
   try {
-    token = await BetterSocialCore.user.createUser(insertedObject?.user);
-    anonymousToken = Getstream.core.createToken(
-      insertedObject?.anonymousUser?.user_id?.toLowerCase()
-    );
-    await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    if (isUnverifiedUser) {
+      token = Getstream.core.createToken(insertedObject?.user?.user_id);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+    } else {
+      token = await BetterSocialCore.user.createUser(insertedObject?.user);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+      await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to my previous comment, the error handling in this section could be improved by providing a more specific log message.

```diff
   } catch (e) {
     console.log('error on inserting user to getstream creating', e);
+    console.log('error on creating user in Getstream', e);
     return res.status(500).json({
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->